### PR TITLE
Fixed input autoFocus not working inside Dropdown

### LINKED ISSUE #1082: Autofocus is not working for Input component if the Input is placed inside Dropdown component
Same issue exist for Textarea component also.
Sandbox link: https://codesandbox.io/s/red-surf-vwtrr0?file=/src/App.js
Use case in neetoPlanner: https://github.com/bigbinary/neeto-planner-web/issues/1209

cc // @amaldinesh7 @rishavkr08

### PR DESCRIPTION
**Pull request by mohit-s96**
closes #1082

### CODE REVIEW

**Comment by mohit-s96** [comment]
@karthiknmenon _a. Please review.

**Comment by mohit-s96** [comment]
@ajmaln _a. Please review.

**Comment by amaldinesh7** [comment]
@ajmaln _a I don't think we need to add this fix. This issue can be resolved from the project end and it's a minor use case.  Thoughts?

File: lib/components/Dropdown.js [hunk 1]
```diff
@@ -133,17 +134,25 @@ const Dropdown = ({
         modifiers: dropdownModifiers,
       }}
       maxWidth="none"
+      onShow={() => {
+        setMounted(true);
+      }}
```
Comment (ajmaln) [inline lib/components/Dropdown.js:139]: Use `onMount` instead of `onShow`

File: lib/components/Input.js [hunk 2]
```diff
@@ -40,6 +42,10 @@ const Input = forwardRef((props, ref) => {
   const valueLength = value?.toString().length || 0;
   const { maxLength } = otherProps;
 
+  useEffect(() => {
+    otherProps.autoFocus && ref.current?.focus();
+  }, [ref.current]);
+
```
Comment (ajmaln) [inline lib/components/Input.js:48]: You can remove this since it will get focused on mount

File: lib/components/Input.js [hunk 3]
```diff
@@ -29,6 +29,8 @@ const Input = forwardRef((props, ref) => {
   const [valueInternal, setValueInternal] = useState(props.value);
   const id = useId(props.id);
 
+  ref = ref || useRef();
+
```
Comment (ajmaln) [inline lib/components/Input.js:33]: Remove this as well

**Review by ajmaln — COMMENTED** [review]
@mohit-s96 _a Please check the suggestions

**Comment by mohit-s96** [comment]
@ajmaln _a. Please review the changes.

**Comment by amaldinesh7** [comment]
@mohit-s96 _a Please add the changelog.

**Review by ajmaln — APPROVED** [review]

**Comment by ajmaln** [comment]
@amaldinesh7 _a Please review and merge

**Pull request merged**